core/src/main/scala-3/chisel3/FormalContractMacro.scala

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3
+
+import scala.quoted.*
+
+object FormalContractMacro {
+
+  /** A macro for `FormalContract` that allows for contracts to be built while
+    * preserving the types of individual elements passed to the contract. This
+    * allows the types of the contract's results to match the type of the
+    * arguments passed to the contract.
+    *
+    * A call to `FormalContract(a: T)` becomes:
+    * ```
+    * FormalContract.apply(
+    *   Seq(a),
+    *   (values => values(0).asInstanceOf[T])
+    * )
+    * ```
```

Comment:
Does `` ``` `` work in Scala3? This used to have to be `{{{` and `}}}` (as the PR's later Scala-generic examples do)?

Reply:
My bad, probably should be `{{{` and `}}}`. Used to the Markdown docs of Rust and Clang.